packages/amplify_api_plugin_interface/lib/src/types/pagination/model_pagination.dart

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+class ModelPagination {
```

Comment:
callout: Do we still want this class? It makes the customer-facing API a little verbose just to add a limit, also adds files/cost/weight/electricity.

Reply:
It should also have a `nextToken` field, so I think that might be why it was originally a class maybe?

Reply:
There was some discussion about that. For now, I think I will leave this as-is and we can have the discussion about this and/or exposing the nextToken param in the PR for the pagination helpers as it will be easier to have that discussion with the context that the PR will provide.